pkg/commands/plan/plan.go

```diff
@@ -4,61 +4,281 @@
+		}); err != nil {
+		return &Result{commentDetails: stderr.String()}, fmt.Errorf("failed to initialize: %w", err)
+	}
+	stderr.Reset()
+	c.actions.EndGroup()
+
+	c.actions.Group("Validating Terraform")
+
+	if _, err := c.terraformClient.Validate(
+		ctx,
+		c.Stdout(),
+		multiStderr,
+		&terraform.ValidateOptions{NoColor: util.Ptr[bool](true)},
+	); err != nil {
+		return &Result{commentDetails: stderr.String()}, fmt.Errorf("failed to validate: %w", err)
+	}
+
+	stderr.Reset()
+	c.actions.EndGroup()
+
+	c.actions.Group("Running Terraform plan")
+
+	planExitCode, err := c.terraformClient.Plan(ctx, c.Stdout(), multiStderr, &terraform.PlanOptions{
+		Out:              util.Ptr[string](c.planFilename),
+		Input:            util.Ptr[bool](false),
+		NoColor:          util.Ptr[bool](true),
+		DetailedExitcode: util.Ptr[bool](true),
+		LockTimeout:      util.Ptr[string](c.flagLockTimeout.String()),
+	})
+
+	stderr.Reset()
+	c.actions.EndGroup()
+
+	hasChanges := planExitCode == 2
+
+	if err != nil && !hasChanges {
+		return &Result{commentDetails: stderr.String()}, fmt.Errorf("failed to plan: %w", err)
+	}
+
+	if !hasChanges {
+		return &Result{hasChanges: hasChanges}, nil
+	}
+
+	c.actions.Group("\n\nFormatting plan output")
+
+	_, err = c.terraformClient.Show(ctx, multiStdout, multiStderr, &terraform.ShowOptions{File: util.Ptr[string](c.planFilename), NoColor: util.Ptr[bool](true)})
+	if err != nil {
+		return &Result{
+			commentDetails: stderr.String(),
+			hasChanges:     hasChanges,
+		}, fmt.Errorf("failed to terraform show: %w", err)
+	}
+
+	stderr.Reset()
+	c.actions.EndGroup()
+
+	githubOutput := terraform.FormatOutputForGitHubDiff(stdout.String())
+
+	planFilePath := path.Join(c.flagWorkingDirectory, c.planFilename)
+
+	planData, err := os.ReadFile(planFilePath)
+	if err != nil {
+		return &Result{
+				hasChanges: hasChanges,
+			},
+			fmt.Errorf("failed to read plan binary: %w", err)
+	}
+
+	c.actions.Group("Uploading plan file")
+
+	if err := c.handleUploadGuardianPlan(ctx, planFilePath, planData, planExitCode); err != nil {
+		return &Result{hasChanges: hasChanges}, fmt.Errorf("failed to upload plan data: %w", err)
+	}
+
+	stderr.Reset()
+	c.actions.EndGroup()
+
+	return &Result{
+		commentDetails: githubOutput,
+		hasChanges:     hasChanges,
+	}, nil
 }
 
-func (p *PlanCommand) Run(ctx context.Context, args []string) error {
-	f := p.Flags()
-	if err := f.Parse(args); err != nil {
-		return fmt.Errorf("failed to parse flags: %w", err)
+// handleUploadGuardianPlan uploads the Guardian plan binary to the configured Guardian storage bucket.
+func (c *PlanCommand) handleUploadGuardianPlan(ctx context.Context, planFilePath string, planData []byte, planExitCode int) error {
+	guardianPlanName := fmt.Sprintf("guardian-plans/%s/%s/%d/%s", c.cfg.RepositoryOwner, c.cfg.RepositoryName, c.cfg.PullRequestNumber, planFilePath)
+
+	metadata := make(map[string]string)
+	metadata["plan_exit_code"] = fmt.Sprintf("%d", planExitCode)
```

Comment:
```suggestion
	metadata["plan_exit_code"] = strconv.FormatInt(planExitCode, 10)
```

Reply:
Needs to be int not int64, used strconv.Itoa()